### PR TITLE
Add Storybook 7 renderer/preset/framework

### PR DESCRIPTION
**Pull request by ThaNarie**
- Remove old and legacy packages.
- Move server rendering inside storybook middleware.
- Update to full ESM

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1 @@
+packages/*/dist
```
Comment (leroykorterink) [inline .prettierignore]:
```suggestion
packages/*/dist

```

Add new line

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-16
\ No newline at end of file
+18
```
Comment (leroykorterink) [inline .nvmrc]:
```suggestion
18

```

Add new line

File: packages/example/.nvmrc [hunk 3]
```diff
@@ -1 +1 @@
-16
\ No newline at end of file
+18
```
Comment (leroykorterink) [inline packages/example/.nvmrc]:
```suggestion
18

```

Add new line

File: packages/example/.storybook/main.js [hunk 4]
```diff
@@ -1,17 +1,24 @@
...
-  },
+  // core: {
+  //   builder: {
+  //     name: '@storybook/builder-webpack5',
+  //     options: {
+  //       fsCache: true,
+  //       lazyCompilation: true,
+  //     },
+  //   },
+  // },
```
Comment (leroykorterink) [inline packages/example/.storybook/main.js]:
```suggestion
```

Remove commented out configuration

File: packages/example/.storybook/main.js [hunk 5]
```diff
@@ -1,17 +1,24 @@
...
+      files: '**/*.stories.@(js|jsx|ts|tsx)',
       titlePrefix: '',
     },
   ],
-  addons: ['@storybook/addon-links', '@storybook/addon-essentials', '@storybook/addon-postcss'],
+  addons: [
+    '@storybook/addon-links',
+    '@storybook/addon-essentials' /*, '@storybook/addon-postcss'*/,
+  ],
   // staticDirs: ['../public'],
```
Comment (leroykorterink) [inline packages/example/.storybook/main.js]:
```suggestion
    '@storybook/addon-essentials'
  ],
```

Remove commented out configuration

File: packages/example/src/components/toggle-expand/toggle-expand.html.twig [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    <p class="mb-0">When clicking the buttons, the value should update accordingly.</p>
+  </div>
+  <div>Value: <span data-ref="label" class="badge rounded-pill bg-primary"></span></div>
+  <div>data: {{ data }}</div>
+  <div style="margin-top: 20px">
+    <button type="button" data-ref="btnToggle" class="btn btn-primary" onToggle="{{ onToggle }}">Toggle</button>
+    <button type="button" data-ref="btnEnable" class="btn btn-success">Enable</button>
+    <button type="button" data-ref="btnDisable" class="btn btn-danger">Disable</button>
+  </div>
+</div>
```
Comment (leroykorterink) [inline packages/example/src/components/toggle-expand/toggle-expand.html.twig]:
```suggestion
</div>

```

Add new line

File: packages/framework-webpack/.eslintignore [hunk 7]
```diff
@@ -0,0 +1,2 @@
+dist/**/*
+preset.js
```
Comment (leroykorterink) [inline packages/framework-webpack/.eslintignore]:
```suggestion
preset.js

```

Add new line

File: packages/framework-webpack/.nvmrc [hunk 8]
```diff
@@ -0,0 +1 @@
+18
```
Comment (leroykorterink) [inline packages/framework-webpack/.nvmrc]:
```suggestion
18

```

Add new line

File: packages/preset-webpack/.nvmrc [hunk 9]
```diff
@@ -0,0 +1 @@
+18
```
Comment (leroykorterink) [inline packages/preset-webpack/.nvmrc]:
```suggestion
18

```

Add new line

File: packages/renderer/.nvmrc [hunk 10]
```diff
@@ -0,0 +1 @@
+18
```
Comment (leroykorterink) [inline packages/renderer/.nvmrc]:
```suggestion
18

```

Add new line

File: packages/framework-webpack/package.json [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+  "files": [
+    "dist/**/*",
+    "README.md",
+    "*.js",
+    "*.d.ts"
+  ],
+  "scripts": {
+    "dev": "npm run build -- --watch",
+    "build": "tsc -p ./tsconfig.json",
+    "typecheck": "tsc --project tsconfig.json --noEmit --noUnusedLocals",
```
Comment (leroykorterink) [inline packages/framework-webpack/package.json:47]: Is there a reason to only enable `--noUnusedLocals` in the typecheck command?

File: packages/renderer/package.json [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+  },
+  "dependencies": {
+    "@muban/muban": "^1.0.0-alpha.35",
+    "@muban/template": "^1.0.0-alpha.4",
+    "@storybook/core-client": "^7.0.7",
+    "@storybook/docs-tools": "^7.0.7",
+    "@storybook/global": "^5.0.0",
+    "@storybook/preview-api": "^7.0.7",
+    "@storybook/types": "^7.0.7",
+    "qs": "^6.11.1",
```
Comment (leroykorterink) [inline packages/renderer/package.json:64]: You're using this package just once and I think you can change the implementation of that code so that it's not necessary to use this package at all.

```ts
  const fetchUrl = getUrl(
    url,
    path,
    new URLSearchParams([
      ...Object.entires(storyContext.globals),
      ...Object.entries(parameters),
    ])
  );
```

```ts
> new URLSearchParams([['kaas', 'ham'], ['milk', 'water']]).toString()
> kaas=ham&melk=water
```

Can't add a suggestion on the code that uses `qs` because it hasn't changed. But I think we can change it since the `qs` package version is change here. It's also fine to do it in another PR.

**Review by leroykorterink — CHANGES_REQUESTED** [review]

File: packages/framework-webpack/package.json (hunk 11 above)
Comment (ThaNarie) [reply packages/framework-webpack/package.json:47]: This has been done for a while in all setups, and I think it was done that way to don't show errors in your IDE or build when you are busy writing code (and you might end up with unused vars temporarily), but do have it there when you typecheck / commit.

Alternatively you could have an extra tsconfig.json, but for just this flag (or two) this seems managable.

File: packages/renderer/package.json (hunk 12 above)
Comment (ThaNarie) [reply packages/renderer/package.json:64]: URLSearchParams doesn't work with nested structures unfortunately.
